spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/RemoveCarryoverIterator.java

```diff
@@ -151,13 +153,21 @@ private int[] generateIndicesToIdentifySameRow(int columnSize) {
     return indices;
   }
 
-  private boolean isSameRecord(Row currentRow, Row nextRow) {
-    for (int idx : indicesToIdentifySameRow) {
+  protected boolean isSameRecord(Row currentRow, Row nextRow) {
+    for (int idx : indicesToIdentifySameRow()) {
```

Comment:
Good catch. Removed it.

Reply:
There are different `indicesToIdentifySameRow` for the two iterators. We need to exclude the index of meta columns `CHANGE_ORDINAL` and `COMMIT_SNAPSHOT_ID` in case of net changes.